packages/unpaywall/tests/unpaywall.spec.js

```diff
@@ -0,0 +1,34 @@
+describe("'primo-explore-unpaywall' package smoke tests", function () {
+  beforeEach(module("bulibUnpaywall"));
+  beforeEach(function(){ jasmine.getJSONFixtures().fixturesPath = "base/tests/fixtures"; })
+
+  var $componentController, $http, bindings, locals, unpaywallController, prmSearchResultAvailabilityLine;
+  beforeEach(
+    inject(function ($injector) {
+      // get/prepare helpers needed to grad/set the controller
+      $componentController = $injector.get("$componentController");
+      $http = $injector.get("$http");
+      prmSearchResultAvailabilityLine = {};
+      prmSearchResultAvailabilityLine.result = getJSONFixture("online_result.json");
+
+      // grab the unpaywallController via [$componentController](https://docs.angularjs.org/api/ngMock/service/$componentController)
+      locals = null;
+      bindings = { 
+        $http: $http, 
+        $injector: $injector,
+        prmSearchResultAvailabilityLine: prmSearchResultAvailabilityLine 
+      };
+      unpaywallController = $componentController('unpaywallController', locals, bindings);
```

Comment:
@atla5 Here, the `$componentController` service is being asked to locate a component/directive named `unpaywallController`, which does not exist. You probably want something like this instead: 

```javascript
// locate the component named 'bulibUnpaywall' and return its controller
unpaywallController = $componentController('bulibUnpaywall', locals, bindings);
```

Reply:
gotcha. thank you! i was trying to target [the following](https://github.com/bulib/primo-explore-bu/blob/master/packages/unpaywall/src/unpaywall.module.js#L2), but this ⤴️ did the trick. i appreciate it!

```js
.controller('unpaywallController', ['$http', '$injector',
    function($http, $injector) { ... }
)
```